### ISSUE TITLE
Crash Bug for 1.0.3.2

### ISSUE THREAD

**Issue by rayman3003** [erickutcher/httpdownloader#139]
Drag & dropping links into "URL Drop windows", would makes the app to crash. It might not crash for the first few drops. But after 4-5 links drops or sometimes a little more, It crashes for sure.

Crash happens only for 1.0.3.2 version.

This problem not occurs for previous 1.0.3.1 version.

**Comment by erickutcher** [comment]
You'll have to give me some more info because I'm not able to reproduce it.

Let me know which version you're using (32bit or 64bit) and your current operating system.

Give me a list of links that are guaranteed to cause the crash.

After you dropped the URL into the drop window, did you close the Add URL(s) window or continue to drop URLs while it was opened?

**Comment by rayman3003** [comment]
> After you dropped the URL into the drop window, did you close the Add URL(s) window or continue to drop URLs while it was opened?

No. I set it to start download immediately after dropping the link, without showing the add url window.

I'm using x64 version of http downloader on windows 10 pro x64 build 1909.

--------------------------

I don't know why, But I can't find the exact links that made this crash, So I can't give u the exact links to reproduce it. 

But I fell to another crash:

Here is my settings. Use these settings (and use as portable mode):

[http_downloader_settings.zip](https://github.com/erickutcher/httpdownloader/files/5086411/http_downloader_settings.zip)

Add all of these links to http downloader & start download. some seconds after starting the download, select all links in the download queue & select "remove & delete". And it starts to crash!

https://0bin.net/paste/MXssFOKj#4DIEC0ZrBCSQSboyvex-CG6fd89Mei7Kh/3Zl/3dG6k

**Comment by erickutcher** [comment]
I've fixed the remove/remove and delete crash. The first crash I'm still looking into. Let me know if you can get it to crash consistently with certain URLs.

**Comment by rayman3003** [comment]
> 
> 
> I've fixed the remove/remove and delete crash. The first crash I'm still looking into. Let me know if you can get it to crash consistently with certain URLs.

Thank u very much. (∩▂∩)

I will report back here, ASAP If I see the crash again.
